vuu-ui/packages/vuu-layout/src/layout-persistence/LayoutPersistenceManager.ts

```diff
@@ -0,0 +1,58 @@
+import { LayoutJSON } from "@finos/vuu-layout";
+import { Layout, LayoutMetadata } from "@finos/vuu-shell";
+
+export interface LayoutPersistenceManager {
+  /**
+   * Saves a new layout
+   *
+   * @param metadata - Metadata about the layout to be saved (excluding ID)
+   * @param layout   - Full JSON representation of the layout to be saved
+   *
+   * @returns ID assigned to the saved layout
+   */
+  saveLayout: (layouts: Layout[]) => string;
+
+  /**
+   * Overwrites an existing layout with a new one
+   *
+   * @param id       - Unique identifier of the existing layout to be updated
+   * @param metadata - Metadata describing the new layout to overwrite with (excluding ID)
+   * @param layout   - Full JSON representation of the new layout to overwrite with
+   */
+  updateLayout: (id: string, metadata: Omit<LayoutMetadata, "id">, layout: LayoutJSON) => void;
+
+  /**
+   * Deletes an existing layout
+   *
+   * @param id - Unique identifier of the existing layout to be deleted
+   */
+  deleteLayout: (id: string) => void;
+
+  /**
+   * Retrieves an existing layout
+   *
+   * @param id - Unique identifier of the existing layout to be deleted
+   *
+   * @returns the layout corresponding to provided metadata
+   */
+  loadLayout: (id: string) => LayoutJSON;
+
+  /**
+   * Retrieves metadata for all existing layouts
+   *
+   * @returns an array of all persisted layout metadata
+   */
+  loadMetadata: () => LayoutMetadata[];
+
+  /**
+   * Retrieves metadata for all existing layouts created by a given user
+   *
+   * @param user - Name of user
+   *
+   * @returns an array of metadata describing all persisted layouts created by {@link user}
+   */
+  loadMetadataByUser: (user: string) => LayoutMetadata[];
+
+  // TODO: should be switched over to load metadata only; remove this
+  loadLayouts: () => Layout[];
```

Comment:
In my mind, we don't really want to be loading a list of layouts at any point. When populating the "My Layouts" list we only need metadata; when one is selected, we should then retrieve the actual data for that layout only. However, all this relies on the layouts being independent (one per tab), which I think will be the case after the JSON resfactor/restructure.

Reply:
Hmmm yeah I see what you're saying.

Even in the case where we know loading _all_ layouts _should_ never be required, I'd favour keeping the naming explicit to what it's doing in case we wanted to introduce that functionality in the future. So I'd have `loadLayouts()` and `loadLayoutsMetadataOnly()`.

Reply:
Clarified offline that this is just going to be removed.